pkg/infrastructure/clusterapi/clusterapi.go

```diff
@@ -189,25 +189,21 @@ func (i *InfraProvider) Provision(ctx context.Context, dir string, parents asset
 	logrus.Infof("Waiting up to %v (until %v %s) for network infrastructure to become ready...", timeout, untilTime.Format(time.Kitchen), timezone)
 	var cluster *clusterv1.Cluster
 	{
-		if err := wait.ExponentialBackoffWithContext(ctx, wait.Backoff{
-			Duration: time.Second * 10,
-			Factor:   float64(1.5),
-			Steps:    32,
-			Cap:      timeout,
-		}, func(ctx context.Context) (bool, error) {
-			c := &clusterv1.Cluster{}
-			if err := cl.Get(ctx, client.ObjectKey{
-				Name:      clusterID.InfraID,
-				Namespace: capiutils.Namespace,
-			}, c); err != nil {
-				if apierrors.IsNotFound(err) {
-					return false, nil
+		if err := wait.PollUntilContextTimeout(ctx, 15*time.Second, timeout, true,
```

Comment:
This line and the one like it below are the only actual change. The rest is indenting.

Reply:
How about we use jittered backoff? I feel we need a backoff otherwise LGTM
https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#JitterUntilWithContext

Reply:
That would be a better alternative than exponential, but I don't think we need backoff at all. The installer is the only client for the control plane and the polling is very sparse, once every 15 seconds, so  there would be no issues with overloading the control plane or rate-limiting.